src/nrniv/nrnbbcore_write.cpp

```diff
@@ -749,6 +751,9 @@ void datumtransform(CellGroup* cgs) {
         datumindex_fill(ith, cg, di, ml);
       }
     }
+    if (corenrn_direct == false && cg.group_id < 0 && (cg.ntype > 0 || cg.n_output > 0)) {
```

Comment:
```suggestion
    // if model is being transferred via files :  if the gid used for filename is < 0 and if there are are
    // real cells in the NrnThread then throw an error
    if (corenrn_direct == false && cg.group_id < 0 && (cg.ntype > 0 || cg.n_output > 0)) {
```

@nrnhines : could you remind me why we check `cg.ntype > 0`?

Reply:
When I added a single empty section to the ranks (in addition to the single cell in rank 1) and printed ntype and n_output before the execerror, I noticed that n_output was 1. And  when adding an ARTIFICIAL_CELL (with no gid), the ntype was 1. I tried various uncommented out variants to ringtest.py of the form
```
diff --git a/ringtest.py b/ringtest.py
index 9af9b33..967b287 100644
--- a/ringtest.py
+++ b/ringtest.py
@@ -146,6 +146,9 @@ if __name__ == '__main__':
     types = shuffle([i % ntype for i in range(ncell * nring)], ran)
     rings = [Ring(ncell, nbranch, ncompart, i * ncell, types) for i in range(nring)]
 
+    #soma = h.Section(name='soma')
+    #acell = h.IntFire1()
+
     timeit("created rings", settings.rank)
 
     # randomize parameters if asked
```

Reply:
Ok thanks for explanation! could you just update the comment I have suggested about with this info i.e. ntype > 0 for art cells with empty section.

Reply:
I'm pushing a change with a comment similar to your suggestion. With a bit more looking, I see that ```CellGroup.n_mech > 0``` handles both the ARTIFICIAL_CELL without gid and empty Section without gid case.